addons/mail/models/mail_mail.py

```diff
@@ -315,16 +316,42 @@ def _prepare_outgoing_body(self):
             return ''
         return self.env['mail.render.mixin']._replace_local_links(self.body_html)
 
-    def _prepare_outgoing_list(self):
+    def _personalize_outgoing_body(self, body, partner=False, records_followers_set=None):
+        """ Return a modified body based on the recipient (partner).
+
+        It must be called when using standard notification layouts
+        even for message without partners.
+
+        :param str body: body to personalize for the recipient
+        :param Model partner: recipient
+        :param set records_followers_set: see ``Followers._get_mail_followers()``
+        """
+        self.ensure_one()
+
+        if (records_followers_set and '/mail/unfollow' in body and partner and
```

Comment:
Actually “records_followers_set” (renamed recipients_follower_status) maybe None if there are no mail with unfollow link. So yes, we could get rid of this condition as we also check that the link is in the body. But that is only possible because we have analysed every call to that method and deduced that. The method can accept None for that parameter and it is also the default value. So is it not better to check it is not None before use it ?